db/event_manager_test.go

```diff
@@ -446,8 +468,8 @@ func TestWebhookBasic(t *testing.T) {
 		em.RaiseDocumentChangeEvent(bodyBytes, docId, "", channels)
 	}
 
-	time.Sleep(50 * time.Millisecond)
-	assert.Equal(t, 4, wr.GetCount())
+	err = wr.waitForCount(context.TODO(), 4, base.DefaultWaitForWebhook)
```

Comment:
If 5 events fire, the test should fail.  This approach doesn't cover that.

This is a flaw in the EventManager design - it's not well-designed for testability.  However, it feels like there are probably a couple of options without rewriting it:
- Add stats for scheduled and completed events, and use those as wait conditions
- Have one (potentially slow-running) test to validate the async aspects of the EventManager asyncEventChannel/processEvent code, and rework the rest of the tests to make synchronous calls to HandleEvent.

Reply:
I had a look at the idea of adding stats but the EventManager doesn't have any hook into db which it'd need for stats right?

Reply:
Storing standalone atomic counters in the `EventManager` seems reasonable for the purposes of testing. We don't need to expose them as expvar stats.